app/models/frontend/nics/NICsAnswers.scala

```diff
@@ -16,24 +16,30 @@
 
 package models.frontend.nics
 
+import cats.implicits.catsSyntaxOptionId
 import models.connector.api_1639.SuccessResponseAPI1639
 import models.database.nics.NICsStorageAnswers
+import models.error.ServiceError
 import play.api.libs.json.{Format, Json}
 
-case class NICsAnswers(class2NICs: Boolean)
+case class NICsAnswers(class2Answers: Option[NICsClass2Answers], class4Answers: Option[NICsClass4Answers])
 
 object NICsAnswers {
   implicit val formats: Format[NICsAnswers] = Json.format[NICsAnswers]
 
-  def mkPriorData(maybeApiAnswers: Option[SuccessResponseAPI1639], maybeDbAnswers: Option[NICsStorageAnswers]): Option[NICsAnswers] = {
+  def invalidAnswersError(answers: NICsAnswers): ServiceError = ServiceError.ErrorFromUpstream(
```

Comment:
It is better to create a specific final case class for your NICS error, so the error message is there:

```
  final case class InvalidNICsAnswer(answers: NICsAnswers) extends ServiceError {
    val errorMessage: String = s"n---------------------\nNICsAnswers must contain only one of 'class2Answ...."
    override val status: Int = BAD_REQUEST
  }
```